dvc/hash_info.py

```diff
@@ -7,6 +7,7 @@
 class HashInfo:
     name: str
     value: str
+    dir_info: list = field(default=None, compare=False)
```

Comment:
Is it possible to have a non-nullable HashInfo, i.e. without `name` and `value` being None?
I understand that it's just a simple container, but I want to avoid possibilities of `hash_info` being:
1. `None`
2. A regular hash_info
3. hash_info with `value=None`.

1 and 2 are fine, it'd be safer to avoid 3rd case if we can.

Reply:
> Is it possible to have a non-nullable HashInfo, i.e. without name and value being None?

@skshetry it is not, but we allow it. Very good points! I also thought about it, but decided not to mess with it for now. Ideally thinking about making this dataclass read-only, so everything will be set when creating an instance and so we can put appropriate checks/asserts for such things